sympy/solvers/solveset.py

```diff
@@ -209,6 +209,32 @@ def _invert_complex(f, g_ys, symbol):
                                                log(Abs(g_y))), S.Integers)
                                for g_y in g_ys])
             return _invert_complex(f.args[0], exp_invs, symbol)
+
+    if isinstance(f, (sin, csc)):
```

Comment:
I don't understand why you have reciprocal functions here. The zeros of sin are not the same as the zeros of csc. By the way, in this branch, `solveset_complex(csc(x), x)` gives `ZeroDivisionError`.